File: Python Program/lab3.c
/*Stop and wait protocol*/

#include<stdio.h>

int main(){
    int framesize,sent=0,ack,i;
    printf("Enter number of frames : ");
    scanf("%d",&framesize);
    while(1){
        for(i=0;i<framesize;i++){
            printf("from %d has been transmitted.\n",sent);
            sent++;
            if(sent==framesize)
            break;
        }
        printf("\nPlease enter the last acknowledgement recieved.\n");
        scanf("%d",&ack);
        if(ack==framesize)
        break;
        else
        sent=ack;
    }
    return 0;
}

File: Python Program/lab3.py
framesize = int(input("Enter number of frames : "))
sent = 0
while True:
    for i in range(framesize):
        print("from", sent, "has been transmitted.")
        sent += 1
        if sent == framesize:
            break
    ack = int(input("\nPlease enter the last acknowledgement received.\n"))
    if ack == framesize:
        break
    else:
        sent = ack
